app/src/main/java/org/simple/clinic/recentpatient/AllRecentPatientsEffectHandler.kt

```diff
@@ -40,10 +42,12 @@ class AllRecentPatientsEffectHandler @AssistedInject constructor(
       effects
           .observeOn(schedulersProvider.io())
           .switchMap {
-            patientRepository
-                .recentPatients_old(currentFacility.get().uuid)
-                .map(::RecentPatientsLoaded)
+            pagerFactory.createPager(
+                sourceFactory = { patientRepository.recentPatients(facilityUuid = currentFacility.get().uuid) },
```

Comment:
Curious, but what thread does the lambda passed as the `sourceFactory` argument run on? `currentFacility.get()` will perform a database call, so I just want to be sure that we are not falling into a main thread IO crash here.

Reply:
So everything that is run inside the `switchMap` should run on whatever thread we set in `observeOn` right? Since it's downstream. At least from my testing, I didn't receive any IO usage on main thread crashes. But I can move it outside the factory as a variable just to be safe.

Reply:
Not really necessary, just wanted to confirm that the `Pager` class does not attempt to change the thread or something when invoking this.

Reply:
I made that change just to be safe anyway 😂